src/imp/libc/fs/syscalls.rs

```diff
@@ -116,13 +116,41 @@ use {
 #[cfg(not(target_os = "redox"))]
 use {super::super::offset::libc_openat, crate::fs::AtFlags};
 
+#[cfg(any(target_os = "android", target_os = "linux"))]
+fn openat_via_syscall(
+    dirfd: BorrowedFd<'_>,
+    path: &ZStr,
+    oflags: OFlags,
+    mode: Mode,
+) -> io::Result<OwnedFd> {
+    unsafe {
+        let dirfd = borrowed_fd(dirfd);
+        let path = c_str(path);
+        let oflags = oflags.bits();
+        let mode = c::c_uint::from(mode.bits());
+        ret_owned_fd(libc::syscall(
+            libc::SYS_openat,
+            dirfd as c::c_long,
+            path as c::c_long,
+            oflags as c::c_long,
+            mode as c::c_long,
+        ) as c::c_int)
+    }
+}
+
 #[cfg(not(target_os = "redox"))]
 pub(crate) fn openat(
     dirfd: BorrowedFd<'_>,
     path: &ZStr,
     oflags: OFlags,
     mode: Mode,
 ) -> io::Result<OwnedFd> {
+    // Work around https://sourceware.org/bugzilla/show_bug.cgi?id=17523
+    // Basically old glibc versions don't handle O_TMPFILE correctly.
+    #[cfg(any(target_os = "android", target_os = "linux"))]
```

Comment:
(I guess since we know android doesn't use glibc, we could just make this `target_os = "linux"`?)

Reply:
Would `target_env = "gnu"` work here? That should properly exclude this code for android, and musl too, which presumably doesn't have this bug.

Reply:
Makes sense.  Pushed that change...and then realized after I did so that it'd likely apply to `x86_64-pc-windows-gnu` and indeed, that build now fails)

So, pushed another fix to also require `unix`.